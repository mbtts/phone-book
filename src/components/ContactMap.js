import { Gmaps, InfoWindow, Marker } from "react-gmaps";
import React, { PureComponent } from "react";
import { geoCode, initialiseMapsApi } from "../api/geocode";

import PropTypes from "prop-types";

class ContactMap extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
  }

  async componentDidMount() {
    await initialiseMapsApi();
    const coords = await geoCode(this.props.contact.address);
    this.setState({ coords });
  }

  render() {
    const map = this.state.coords ? (
      <Gmaps
        height={"30vh"}
        lat={this.state.coords.lat}
        lng={this.state.coords.lng}
        zoom={17}
        loadingMessage={"Loadingâ€¦"}
      >
        <Marker
          lat={this.state.coords.lat}
          lng={this.state.coords.lng}
          draggable={true}
        />
      </Gmaps>
    ) : null;

    return <div className="map__container">{map}</div>;
  }
}

ContactMap.propTypes = {
  contact: PropTypes.shape({
    address: PropTypes.string.isRequired
  })
};

export default ContactMap;
