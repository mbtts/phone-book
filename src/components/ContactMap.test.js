import ContactMap from "./ContactMap";
import { Gmaps } from "react-gmaps";
import React from "react";

jest.mock("../api/geocode");

describe("Map of contact", () => {
  it("should render initial state", () => {
    const wrapper = shallow(
      <ContactMap contact={{ address: "17 Anchor Ave, Darwen BB3 0AZ, UK" }} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a map with coordinates", async () => {
    const wrapper = shallow(
      <ContactMap contact={{ address: "17 Anchor Ave, Darwen BB3 0AZ, UK" }} />
    );

    process.nextTick(() => {
      wrapper.update();
      expect(wrapper.find(Gmaps)).toHaveLength(1);
      expect(wrapper.state("coords")).toEqual({
        lat: 53.6980027,
        lng: -2.476654
      });
    });
  });
});
