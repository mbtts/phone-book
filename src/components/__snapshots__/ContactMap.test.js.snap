// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map of contact should render initial state 1`] = `
<div
  className="map__container"
/>
`;
